src/app/cases/creator/case-creator-submit.component.ts

```diff
@@ -44,6 +47,29 @@ export class CaseCreatorSubmitComponent implements OnInit {
     return (sanitizedEditForm: CaseEventData) => this.casesService.validateCase(this.jurisdictionId, this.caseTypeId, sanitizedEditForm);
   }
 
+  private canSaveDraft(): boolean {
+    return this.eventTrigger.can_save_draft === true;
+  }
+
+  private isCreatingDraft(): boolean {
+    return !this.eventTrigger.draft_store_id;
+  }
+
+  saveDraft(): (caseEventData: CaseEventData) => Observable<Draft> {
+    if (this.canSaveDraft()) {
+      if (this.isCreatingDraft()) {
+        return (caseEventData: CaseEventData) => this.draftService.createDraft(this.jurisdictionId,
```

Comment:
I would have thought the Draft store API would provide an upsert method that creates or updates...do we need to do this check here and call either save or update on the draft store?